### ISSUE TITLE
Preload fails with error related to reflection in PHP8.0rc2

### ISSUE THREAD

**Issue by adhocore** [symfony/symfony#38836]
**Symfony version(s) affected**: 5.1.7

```
NOTICE: PHP message: PHP Fatal error:  Uncaught Error: Call to undefined method ReflectionUnionType::isBuiltin() 
in <project>/vendor/symfony/dependency-injection/Dumper/Preloader.php:116
```
**Description**  

**How to reproduce**  
Just trying to preload the framework generated preload file in cache dir.

**Comment by derrabus** [comment]
~~Duplicate of #38693.~~

Wrong link, see below.

**Comment by adhocore** [comment]
while they seem related, probably not duplicate.
this one is for php8 and not related to exception but reflection:
`unknown method ReflectionUnionType::isBuiltin()`

**Comment by nicolas-grekas** [comment]
Wrong link, here is the fix #38699

**Comment by derrabus** [comment]
Right, sorry. 🙈
